undefined: unix.Madvise on macOS Sierra

If I try to go get badger I get the following: 
```
$  go get -v github.com/dgraph-io/badger
github.com/dgraph-io/badger (download)
github.com/AndreasBriese/bbloom (download)
github.com/AndreasBriese/bbloom
github.com/dgraph-io/badger/options
golang.org/x/sys/unix
github.com/dgraph-io/badger/protos
github.com/dgraph-io/badger/y
# github.com/dgraph-io/badger/y
golang/src/github.com/dgraph-io/badger/y/mmap_unix.go:47:9: undefined: unix.Madvise
```

I am running macOS Sierra and go 1.9.

--- a/y/mmap_unix.go
+++ b/y/mmap_unix.go
@@ -20,6 +20,8 @@ package y
 
 import (
 	"os"
+	"syscall"
+	"unsafe"
 
 	"golang.org/x/sys/unix"
 )
@@ -47,5 +49,15 @@ func Madvise(b []byte, readahead bool) error {
 	if !readahead {
 		flags = unix.MADV_RANDOM
 	}
-	return unix.Madvise(b, flags)
+	return madvise(b, flags)
+}
+
+// This is required because the unix package does not support the madvise system call on OS X.
+func madvise(b []byte, advice int) (err error) {
+	_, _, e1 := syscall.Syscall(syscall.SYS_MADVISE, uintptr(unsafe.Pointer(&b[0])),
+		uintptr(len(b)), uintptr(advice))
+	if e1 != 0 {
+		err = e1
+	}
+	return
 }

